src/polycubed/src/rest_server.cpp

```diff
@@ -931,5 +937,157 @@ const std::string &RestServer::getPort() {
+
+    size_t t; //index inside counter_ or gauges_
+   
+      //i.first is the name of the metric
+      // TODO i.second is the struct InfoMetric: we need pathMetric and typeMetric
+     //is the value returned by the jsoncons query
+     
+     auto servicesctrls_names = core.get_servicectrls_names_vector();
+
+
+    auto running_cubes = core.get_names_cubes();
+    
+    // if a cube is in running cubes but is not in all_cubes_and_metrics it means that I need to create his metrics
+    for(auto cube: running_cubes) {
+           if(all_cubes_and_metrics[cube].empty()) {
+             auto serviceName = core.get_cube_service(cube);
+            for (auto& kv: map_metrics[serviceName].counters_map)
+                kv.second.get().Add({{"cubeName", cube}});
+            for (auto& kv: map_metrics[serviceName].gauges_map)
+                kv.second.get().Add({{"cubeName", cube}});
+           }  
+     }
+
+    // if a cube is in all_cube_and_metrics and is not in running cubes: that cube it was deleted so I need to delete all his metrics i.e.
+    // all gauges and counters with label cubeName = that cube
+     for(auto cube: all_cubes_and_metrics) {
+           std::vector<std::string>::iterator it = std::find(running_cubes.begin(), running_cubes.end(), cube.first);
+           if(it == running_cubes.end()) {
+             auto serviceName = cube.second;
+            for (auto& kv: map_metrics[serviceName].gauges_map) {
+               auto& toDelete = kv.second.get().Add({{"cubeName", cube.first}});
+               kv.second.get().Remove(& toDelete);
+             }
+            for (auto& kv: map_metrics[serviceName].counters_map) {
+               auto& toDelete = kv.second.get().Add({{"cubeName", cube.first}});
+               kv.second.get().Remove(& toDelete);
+             }
+          }
+     }
+
+    // at then end we need that all_cubes_and_metrics and running_cubes with equal values
+     all_cubes_and_metrics.clear();
+     for(auto cube: running_cubes)
+       all_cubes_and_metrics[cube] = core.get_cube_service(cube);
+  
+
+    for(auto serviceName: servicesctrls_names)  {
+        ListKeyValues keys{}; //keys used to create the entire json of a cube
+        //every service has a map with the name of the metric and the InfoMetric struct (we need the pathmetric, typemetric, type-operation and a value)
+        auto mapInfoMetricsService = core.get_service_controller(serviceName).get_mapInfoMetrics();
+          // for every service
+              for(auto cubeName: core.get_service_controller(serviceName).get_names_cubes()) {
+                      std::string cubeStr = core.get_service_controller(serviceName).get_management_interface()->get_service()->ReadValue(cubeName, keys).message;
+                      jsoncons::json cubeJson = jsoncons::json::parse(cubeStr);
+
+                      for(auto& i: mapInfoMetricsService) {
+                       //TODO
+                       if(i.second.pathMetric.compare("TODO")!=0) {
```

Comment:
Add something more: what is this "todo"?
(no need to implement this code, but at least say what it should be supposed to do)

Reply:
Done :+1:

Reply:
In reality it was not a classic TODO.  To clarify I added this comment:

I have considered the case where someone defines metrics in the yang of a service but for some reason does not define the value of the path-metric. Just write TODO and Polycube even having the data structures for that metric,  it will not try to retrieve the value.